modmesh/pilot/_svg.py

```diff
@@ -148,191 +148,272 @@ def calc_vertices(self):
+                else:
+                    start_pos[0] = (x_cur + dx)
+                    start_pos[1] = (y_cur + dy)
 
                 current_pos = start_pos
                 i += 2
+
+                # implicit line-to command
+                while i + 1 < len(coords):
+                    dx, dy = coords[i:i+2]
+                    x_cur, y_cur = current_pos[0], current_pos[1]
+
+                    x_end = dx
+                    y_end = dy
+                    if cmd == 'm':
+                        x_end = (x_cur + dx)
+                        y_end = (y_cur + dy)
+
+                    end = Point(x_end, y_end, 0)
+                    sp2d.append(Segment(current_pos, end))
+
+                    current_pos = end
+                    i += 2
+
                 last_control = None
             elif cmd in ('L', 'l'):
-                # line to (x, y):
-                #   L x y
-                #   l dx dy (relative position to current position)
-                x, y = coords[i], coords[i+1]
-
-                if cmd == 'L':  # absolute position
-                    x_end = x
-                    y_end = y
-                else:  # cmd is 'l', relative position
-                    x_end = current_pos[0] + x
-                    y_end = current_pos[1] + y
-
-                end = Point(x_end, y_end, 0)
-                current_pos = end
-
-                # add a Segment to SegmentPad
-                sp2d.append(Segment(current_pos, end))
-                i += 2
-                last_control = None
-            elif cmd == 'C':
-                # draw cubic bezier curve:
-                #   C ctl_x1, ctl_y1, ctl_x2, ctl_y2, endpos_x, endpos_y
-                x_1, y_1, x_2, y_2, x_end, y_end = coords[i:i+6]
-
-                p0 = current_pos
-                p1 = Point(x_1, y_1, 0)
-                p2 = Point(x_2, y_2, 0)
-                p3 = Point(x_end, y_end, 0)
-
-                # add a Curve to CurvePad with 4 control points
-                cp2d.append(p0=p0, p1=p1, p2=p2, p3=p3)
-
-                current_pos = p3
-                last_control = p2
-                i += 6
-            elif cmd == 'c':
-                # draw cubic bezier curve with relative position:
-                #   c ctl_dx1, ctl_dy1, ctl_dx2, ctl_dy2, endpos_dx, endpos_dy
+                # Draw lines from current position
+                #   command: [L|l] (relative position in lowercase command)
+                #   parameter: (dx1, dy1)+
+                while i + 1 < len(coords):
+                    dx, dy = coords[i:i+2]
+                    x_cur, y_cur = current_pos[0], current_pos[1]
+
+                    x_end = dx
+                    y_end = dy
+                    if cmd == 'l':
+                        x_end = x_cur + dx
+                        y_end = y_cur + dy
+
+                    end = Point(x_end, y_end, 0)
+                    sp2d.append(Segment(current_pos, end))
+
+                    current_pos = end
```

Comment:
Set the end point as current position after adding a segment, which fixes incorrect rendering of vertical lines.